docs/router/plugins.mdx

```diff
@@ -0,0 +1,265 @@
+    Reduce end-to-end request time with direct resolver calls
+  </Card>
+  <Card title="Resource Efficiency" icon="server">
+    Reduce infrastructure costs by consolidating services
+  </Card>
+  <Card title="Clean Architecture" icon="diagram-project">
+    Maintain separation of concerns with modular plugin design
+  </Card>
+</CardGroup>
+
+The plugin architecture is built on HashiCorp's battle-tested [go-plugin framework](https://github.com/hashicorp/go-plugin), which powers production systems like Vault and Terraform with millions of deployments worldwide. This provides enterprise-grade reliability, secure process isolation, and proven stability for your GraphQL infrastructure.
+
+## Motivation
+
+The biggest pain point in adopting GraphQL isn't GraphQL itself it's the steep learning curve and fragmented subgraph tooling across languages and platforms.
+
+<CardGroup cols={1}>
+  <Card title="Simplify GraphQL Implementation" icon="wand-magic-sparkles">
+    While everyone understands the value of "one schema, one query," building and maintaining production-grade subgraphs across diverse environments remains challenging. Spec support, runtime performance, and type safety depend on your subgraph framework quality.
+  </Card>
+
+  <Card title="Design with GraphQL, Implement with gRPC" icon="code">
+    Our approach combines GraphQL's schema-first flexibility with gRPC's performance and reliability. Using the same plugin system that powers HashiCorp Vault and Terraform, you can develop subgraphs in any language and run them directly in the Cosmo Router.
+  </Card>
+
+  <Card title="Proto-Based Code Generation" icon="bolt">
+    The strict typing and automatic code generation from Protocol Buffers (proto) definitions provides an immense productivity boost, especially in the era of generative AI. The generated proto-based gRPC code creates a strongly-typed foundation that AI tools can more effectively understand, extend, and modify, reducing development time and errors while ensuring consistent API implementation.
+  </Card>
+</CardGroup>
+
+### Benefits Over Traditional Subgraphs
+
+When compared to traditional standalone subgraphs, gRPC plugins offer substantial technical advantages:
+
+<CardGroup cols={2}>
+  <Card title="Reduced Latency" icon="gauge-high">
+    Eliminates network round-trips between the router and subgraph for faster query execution
+  </Card>
+  <Card title="Simplified Deployment" icon="cloud-arrow-up">
+    Fewer moving parts means simpler deployments and maintenance with all subgraphs running in the router process
+  </Card>
+  <Card title="Reduced Infrastructure Costs" icon="dollar-sign">
+    Consolidate services to reduce overall resource consumption and operational complexity
+  </Card>
+  <Card title="Direct API Bridging" icon="bridge">
+    Easily connect legacy REST APIs, databases, and other services to your graph without building standalone GraphQL services
+  </Card>
+</CardGroup>
```

Comment:
This reads nearly identical to the above set of 4 cards describing advantages of gRPC subgraphs, perhaps consolidate?